### PR TITLE
Refactoring for handling generic classes

### PR DESCRIPTION
**Pull request by VoglSebastian**
Refactoring of CastClassManager + added an interface for a generic class implementation, to change the used implementation of GenericClass

### CODE REVIEW

File: client/src/main/java/org/evosuite/setup/TestCluster.java [hunk 1]
```diff
@@ -55,6 +52,8 @@
  */
 public class TestCluster {
 
+	// Bei FunctionalInterfaces kann der TestCluster auch MethodenReferenzen oder Lambdas einfügen
```
Comment (schweikl) [inline client/src/main/java/org/evosuite/setup/TestCluster.java:55]: A comment that should be removed?

**Review by schweikl — COMMENTED** [review]
LGTM, some minor things that should be addressed, though (see comments)

**Review by schweikl — COMMENTED** [review]
Seems alright now :slightly_smiling_face:

**Review by schweikl — COMMENTED** [review]
LGTM

**Review by schweikl — COMMENTED** [review]
Is there a particular reason why this Test is now deleted?

**Comment by VoglSebastian** [comment]
> Is there a particular reason why this Test is now deleted?

I added this test to verify, that particular methods of TypeUtils do what i assume they do, but i removed the code in which this was assumed. So i deemed this test useless...

**Review by schweikl — COMMENTED** [review]
The changes look good to me

**Comment by schweikl** [comment]
I guess since all comments are addressed, we are now ready to merge. Unless there is something else you wanted to change?

**Pull request merged**